mmcls/datasets/dataset_wrappers.py

```diff
@@ -170,3 +170,41 @@ def __getitem__(self, idx):
 
     def __len__(self):
         return len(self.repeat_indices)
+
+
+@DATASETS.register_module()
+class KFoldDataset:
+
+    def __init__(self,
+                 dataset,
+                 fold=0,
+                 num_splits=5,
+                 test_mode=False,
+                 seed=None):
+        self.dataset = dataset
+        self.CLASSES = dataset.CLASSES
+        self.test_mode = test_mode
+        self.num_splits = num_splits
+
+        length = len(dataset)
+        indices = list(range(length))
+        if isinstance(seed, int):
+            rng = np.random.default_rng(seed)
+            rng.shuffle(indices)
+
+        test_start = length // num_splits * fold
+        test_end = length // num_splits * (fold + 1)
```

Comment:
Current implementation may omit some samples due to rounding.
 
Let's say we have 29 samples split into 5 folds. For the last fold, `test_start = 20` and `test_end = 25`.


```suggestion
        test_start = length * fold // num_splits
        test_end = length * (fold + 1) // num_splits
```

Reply:
Done